File: solutions/typescript/proverb/1/proverb.ts
export function proverb(...words: string[]): string {
  if (words.length === 0) {
    return '';
  }
  
  const result: string[] = [];
  
  // Generate the "For want of X the Y was lost." lines
  for (let i = 0; i < words.length - 1; i++) {
    result.push(`For want of a ${words[i]} the ${words[i + 1]} was lost.`);
  }
  
  // Add the final line
  result.push(`And all for the want of a ${words[0]}.`);
  
  return result.join('\n');
}

File: solutions/java/proverb/1/src/main/java/Proverb.java
class Proverb {
    private final String[] words;
    
    Proverb(String[] words) {
        this.words = words;
    }

    String recite() {
        // Handle empty input
        if (words == null || words.length == 0) {
            return "";
        }
        
        StringBuilder result = new StringBuilder();
        
        // Generate "For want of..." lines for consecutive pairs
        for (int i = 0; i < words.length - 1; i++) {
            result.append("For want of a ")
                  .append(words[i])
                  .append(" the ")
                  .append(words[i + 1])
                  .append(" was lost.\n");
        }
        
        // Add the final concluding line
        result.append("And all for the want of a ")
              .append(words[0])
              .append(".");
        
        return result.toString();
    }
}